code/obj/item/mob_parts/robot_parts.dm

```diff
@@ -1216,6 +1212,257 @@ ABSTRACT_TYPE(/obj/item/parts/robot_parts/leg/right)
+					playsound(src, 'sound/impact_sounds/Generic_Stab_1.ogg', 40, 1)
+					src.UpdateIcon()
+
+		if (iswrenchingtool(W))
+			var/list/actions = list("Do nothing")
+			if(src.check_completion())
+				actions.Add("Finish and Activate the Drone")
+			if(src.ai_interface)
+				actions.Add("Remove the AI Interface")
+			if(src.cell)
+				actions.Add("Remove the Power cell")
+			if(!actions.len)
+				boutput(user, "<span class='alert'>You can't think of anything to do with the frame.</span>")
+				return
+
+			var/action = tgui_input_list(user, "What do you want to do?", "Drone Frame", actions)
+			if (!action)
+				return
+			if (action == "Do nothing")
+				return
+			if (BOUNDS_DIST(src.loc, user.loc) > 0 && !user.bioHolder.HasEffect("telekinesis"))
+				boutput(user, "<span class='alert'>You need to move closer!</span>")
+				return
+
+			switch(action)
+				if("Finish and Activate the Drone")
+					src.finish_drone()
+				if("Remove the AI Interface")
+					src.ai_interface.set_loc( get_turf(src) )
+					src.ai_interface = null
+					src.UpdateIcon()
+				if("Remove the Power cell")
+					src.cell.set_loc( get_turf(src) )
+					src.cell = null
+					src.UpdateIcon()
+			playsound(src, 'sound/items/Ratchet.ogg', 40, 1)
+			src.UpdateIcon()
+			return
+
+	update_icon()
+		src.icon_state = "frame-" + src.shelltypetoapply;
+
+		if(build_step > 0)
+			src.UpdateOverlays(image('icons/mob/hivebot.dmi', "plating-c-" + src.shelltypetoapply, FLOAT_LAYER, 2),"plating")
+		else
+			src.UpdateOverlays(null,"plating")
+
+		if(build_step > 1)
+			src.UpdateOverlays(image('icons/mob/hivebot.dmi', "wires-c-" + src.shelltypetoapply, FLOAT_LAYER, 2),"wires")
+		else
+			src.UpdateOverlays(null,"wires")
+
+		if(build_step > 2)
+			src.UpdateOverlays(image('icons/mob/hivebot.dmi', "radio-c-" + src.shelltypetoapply, FLOAT_LAYER, 3),"radio")
+		else
+			src.UpdateOverlays(null,"radio")
+
+		if(src.ai_interface)
+			src.UpdateOverlays(image('icons/mob/hivebot.dmi', "interface-c-" + src.shelltypetoapply, FLOAT_LAYER, 3),"interface")
+		else
+			src.UpdateOverlays(null,"interface")
+
+		if(src.cell)
+			src.UpdateOverlays(image('icons/mob/hivebot.dmi', "cell-c-" + src.shelltypetoapply, FLOAT_LAYER, 4),"cell")
+		else
+			src.UpdateOverlays(null,"cell")
+
+
+	proc/check_completion()
+		if (build_step == 3)
+			if ((src.ai_interface) && src.cell)
+				return 1
+		return 0
+
+	proc/finish_drone()
+		var/mob/living/silicon/drone/drone = null
+		drone = new /mob/living/silicon/drone(get_turf(src.loc),src,0,src.emagged)
+
+		drone.name = "Drone"
+		drone.real_name = "Drone"
+
+		if (src.ai_interface)
+			if (!(drone in available_ai_shells))
+				available_ai_shells += drone
+			for_by_tcl(AI, /mob/living/silicon/ai)
+				boutput(AI, "<span class='success'>[src] has been connected to you as a controllable drone.</span>")
+			playsound(src, 'sound/weapons/radxbow.ogg', 40, 1)
+			drone.ai_interface = src.ai_interface
+			drone.ai_interface.set_loc(drone)
+			drone.shell = 1
```

Comment:
```suggestion
			drone.shell = TRUE
```
Broadly, check for usages of `1` that should be changed to `TRUE`, and usages of `0` that should be changed to `FALSE`.